v2/tests/log_sink_etw_int/log_sink_etw_int.c

```diff
@@ -0,0 +1,683 @@
+{
+    char property_name[MAX_PROPERTY_NAME_LENGTH];
+    uint8_t property_type;
+    union
+    {
+        char ascii_char_ptr_value[MAX_STRING_VALUE_LENGTH];
+        uint8_t uint8_t_value;
+        int8_t int8_t_value;
+        uint16_t uint16_t_value;
+        int16_t int16_t_value;
+        uint32_t uint32_t_value;
+        int32_t int32_t_value;
+        uint64_t uint64_t_value;
+        int64_t int64_t_value;
+        uint8_t struct_field_count;
+    };
+} PARSED_EVENT_PROPERTY;
+
+typedef struct PARSED_EVENT_TAG
+{
+    char event_name[MAX_EVENT_NAME_LENGTH];
+    char content[MAX_CONTENT_LENGTH];
+    char file[MAX_FILE_LENGTH];
+    char func[MAX_FUNC_LENGTH];
+    int line;
+    uint32_t property_count;
+    PARSED_EVENT_PROPERTY properties[MAX_PROPERTY_COUNT];
+} PARSED_EVENT;
+
+#define MAX_EVENTS 128
+
+static volatile LONG parsed_event_count;
+static PARSED_EVENT parsed_events[MAX_EVENTS];
+static HANDLE process_thread_handle;
+static TRACEHANDLE trace;
+static char logger_name[128];
+static char trace_session_name[128];
+
+// This callback is needed
+// Without this callback events are not properly processed
+static ULONG WINAPI event_trace_buffer_callback(PEVENT_TRACE_LOGFILEA log_file)
+{
+    (void)log_file;
+
+    return TRUE;
+}
+
+static void WINAPI event_trace_record_callback(EVENT_RECORD* pEventRecord)
+{
+    if (IsEqualGUID(&pEventRecord->EventHeader.ProviderId, &provider_guid))
+    {
+        const uint8_t* metadata = NULL;
+
+        LONG current_parsed_event_count = InterlockedAdd(&parsed_event_count, 0);
+        (void)memset(&parsed_events[current_parsed_event_count], 0, sizeof(PARSED_EVENT));
+
+        EVENT_HEADER_EXTENDED_DATA_ITEM* extended_data_items = pEventRecord->ExtendedData;
+
+        for (unsigned short i = 0; i < pEventRecord->ExtendedDataCount; i++)
+        {
+            if (extended_data_items[i].ExtType == EVENT_HEADER_EXT_TYPE_EVENT_SCHEMA_TL)
+            {
+                // This one has the event name
+                // Not really documented, but the Windows source code seems to indicate I can just skip over a few bytes that hold a length and version
+                const char* event_name = (const char*)(uintptr_t)(extended_data_items[i].DataPtr + 4);
+
+                if (event_name != NULL)
+                {
+                    (void)strncpy(parsed_events[current_parsed_event_count].event_name, event_name, sizeof(parsed_events[current_parsed_event_count].event_name));
+                    parsed_events[current_parsed_event_count].event_name[sizeof(parsed_events[current_parsed_event_count].event_name) - 1] = '\0';
+                    metadata = (const uint8_t*)(event_name + strlen(event_name) + 1);
+                }
+
+                break;
+            }
+        }
+       
+        POOR_MANS_ASSERT(pEventRecord->UserData != NULL);
+
+        const char* current_user_data_field = pEventRecord->UserData;
+        if (current_user_data_field != NULL)
```

Comment:
>current_user_data_field [](http://example.com/codeflow?start=12&length=23)

why do we need to evaluate this condition when POOR_MANS_ASSERT is just in line 133 above?

Reply:
Not needed, removed.